Failing tests

There are failures with Python 3.12.

```bash
============================= test session starts ==============================
platform linux -- Python 3.12.1, pytest-7.4.3, pluggy-1.3.0
rootdir: /build/pyutil-3.3.6
collected 69 items                                                             

pyutil/test/current/test_assertutil.py .                                 [  1%]
pyutil/test/current/test_fileutil.py .                                   [  2%]
pyutil/test/current/test_jsonutil.py ...                                 [  7%]
pyutil/test/current/test_mathutil.py ..........                          [ 21%]
pyutil/test/current/test_observer.py ....                                [ 27%]
pyutil/test/current/test_time_format.py ....                             [ 33%]
pyutil/test/current/test_verlib.py .....                                 [ 40%]
pyutil/test/current/test_version_class.py ...                            [ 44%]
pyutil/test/current/json_tests/test_decode.py FF                         [ 47%]
pyutil/test/current/json_tests/test_default.py .                         [ 49%]
[...]
pyutil/test/out_of_shape/test_zlibutil.py .                              [100%]

=================================== FAILURES ===================================
___________________________ TestDecode.test_decimal ____________________________

self = <pyutil.test.current.json_tests.test_decode.TestDecode testMethod=test_decimal>

    def test_decimal(self):
        rval = json.loads('1.1', parse_float=decimal.Decimal)
>       self.assert_(isinstance(rval, decimal.Decimal))
E       AttributeError: 'TestDecode' object has no attribute 'assert_'. Did you mean: 'assertIn'?

pyutil/test/current/json_tests/test_decode.py:11: AttributeError
____________________________ TestDecode.test_float _____________________________

self = <pyutil.test.current.json_tests.test_decode.TestDecode testMethod=test_float>

    def test_float(self):
        rval = json.loads('1', parse_int=float)
>       self.assert_(isinstance(rval, float))
E       AttributeError: 'TestDecode' object has no attribute 'assert_'. Did you mean: 'assertIn'?

pyutil/test/current/json_tests/test_decode.py:16: AttributeError
=============================== warnings summary ===============================
```

--- a/pyutil/test/current/json_tests/test_decode.py
+++ b/pyutil/test/current/json_tests/test_decode.py
@@ -8,10 +8,10 @@ from pyutil import jsonutil as json
 class TestDecode(TestCase):
     def test_decimal(self):
         rval = json.loads('1.1', parse_float=decimal.Decimal)
-        self.assert_(isinstance(rval, decimal.Decimal))
+        self.assertIsInstance(rval, decimal.Decimal)
         self.assertEqual(rval, decimal.Decimal('1.1'))
 
     def test_float(self):
         rval = json.loads('1', parse_int=float)
-        self.assert_(isinstance(rval, float))
+        self.assertIsInstance(rval, float)
         self.assertEqual(rval, 1.0)

--- a/pyutil/test/current/json_tests/test_speedups.py
+++ b/pyutil/test/current/json_tests/test_speedups.py
@@ -10,11 +10,11 @@ class TestSpeedups(TestCase):
         if not encoder.c_encode_basestring_ascii:
             raise SkipTest("no C extension speedups available to test")
         self.assertEqual(decoder.scanstring.__module__, "simplejson._speedups")
-        self.assert_(decoder.scanstring is decoder.c_scanstring)
+        self.assertIs(decoder.scanstring, decoder.c_scanstring)
 
     def test_encode_basestring_ascii(self):
         if not encoder.c_encode_basestring_ascii:
             raise SkipTest("no C extension speedups available to test")
         self.assertEqual(encoder.encode_basestring_ascii.__module__, "simplejson._speedups")
-        self.assert_(encoder.encode_basestring_ascii is
+        self.assertIs(encoder.encode_basestring_ascii,
                           encoder.c_encode_basestring_ascii)

